utam-compiler/src/main/java/utam/compiler/translator/ClassSerializer.java

```diff
@@ -139,6 +156,11 @@ private String getImportStatement(TypeProvider type) {
     Set<String> res = new HashSet<>();
     res.add(getImportStatement(source.getBaseClassType()));
     res.add(getImportStatement(source.getImplementedType().getInterfaceType()));
+    source.getDeclaredElementTypes(false).stream()
+        .map(returnType -> ((TypeUtilities.Element)returnType).getBasicInterfaces())
+        .flatMap(Collection::stream)
+        .collect(Collectors.toSet())
```

Comment:
no need to collect before forEach as my static analyzer keeps reminding me :) (i do that often as well)

Reply:
True, but I want this to be a set, not a list so that duplicates are removed. Will `forEach` handle that properly? I'm not sure it will.

Reply:
ah, good point